### PR TITLE
Produce stream of chunks from system.zookeeper instead of accumulating the whole result

### PR DESCRIPTION
**Pull request by davenger**
### Changelog category (leave one):
- Improvement

### Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
Previously the whole result was accumulated in memory and returned as one big chunk. This change should help to reduce memory consumption when reading many rows from `system.zookeeper`, allow showing intermediate progress (how many rows have been read so far) and avoid hitting connection timeout when result set is big.

### CODE REVIEW

**Comment by robot-ch-test-poll4** [comment]
*This is an automated comment for commit 82caaa744a51ba11e0b3be803956c875b43a6aad with description of existing statuses. It's updated for the latest CI running*

[⏳ Click here](https://s3.amazonaws.com/clickhouse-test-reports/59545/82caaa744a51ba11e0b3be803956c875b43a6aad/ci_running.html) to open a full report in a separate page

<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>CI running</td><td>A meta-check that indicates the running CI. Normally, it's in <b>success</b> or <b>pending</b> state. The failed status indicates some problems with the PR</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/59545/82caaa744a51ba11e0b3be803956c875b43a6aad/ci_running.html">⏳ pending</a></td></tr>
<tbody>
</table>

**Review by SmitaRKulkarni — APPROVED** [review]
Rest all LGTM

File: src/Storages/System/StorageSystemZooKeeper.cpp [hunk 1]
```diff
@@ -471,7 +511,7 @@ void ReadFromSystemZooKeeper::fillData(MutableColumns & res_columns)
     };
     std::vector<ListTask> list_tasks;
     std::unordered_set<String> added;
-    while (!paths.empty())
+    while (!paths.empty() && (max_block_size == 0 || row_count < max_block_size))
```
[the remark this answers is not in the document]
Comment (davenger) [reply src/Storages/System/StorageSystemZooKeeper.cpp]: Good point. I've changed the condition to look at total size in bytes instead of rows

**Comment by Algunenano** [comment]
It seems this PR made 02221_system_zookeeper_unrestricted_like flaky, and now it might duplicate the output: https://github.com/ClickHouse/ClickHouse/issues/60190

**Pull request merged**